libs/vkd3d/swapchain.c

```diff
@@ -48,6 +48,28 @@
 
 #define INVALID_VK_IMAGE_INDEX (~(uint32_t)0)
 
+/*#define VKD3D_SWAPCHAIN_ROBUSTNESS_TEST*/
+#ifdef VKD3D_SWAPCHAIN_ROBUSTNESS_TEST
+#include <stdlib.h>
+static bool VKD3D_SPURIOUS_SUBOPTIMAL(void)
+{
+    unsigned int value = rand() & 15;
+    return value >= 8;
+}
+
+static bool VKD3D_SPURIOUS_SURFACE_LOST(void)
+{
+    unsigned int value = rand() & 15;
+    return value >= 8;
+}
+
+static bool VKD3D_SPURIOUS_OUT_OF_DATE(void)
+{
+    unsigned int value = rand() & 15;
+    return value >= 8;
+}
```

Comment:
not a huge fan of the ifdef soup down below; if we actually want to ship this code, maybe we could do this in present/acquire wrappers?

Reply:
It might be better to just leave that commit out for now. It's trivial to re-add later. Was mostly to demonstrate what I did to test things.

Reply:
Removed commit from PR.